src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs

```diff
@@ -78,15 +76,20 @@ protected override async Task<IOffset> ProduceAsync(RawBrokerEnvelope envelope)
             }
         }
 
-        private byte[] GetPartitioningKey(IEnumerable<MessageHeader> headers)
+        private byte[] GetKafkaMessageKey(MessageHeaderCollection headers)
```

Comment:
Rename method to reflect that the header is removed from the collection, if present.

Reply:
:) I like that you have my same mindset when it goes to self documenting code. I will definitely change it (I couldn't come up with a nice name and I left it alone).

Reply:
Name changed!